File: goodStrategy.ts
import { Strategy } from "./strategy";
import { OneDie } from "./oneDie";
import { State } from "./state";
import { Move } from "./move";

export class GoodStrategy implements Strategy {
  private game: OneDie;
  constructor(game: OneDie) {
    this.game = game;
  }

  getMove(state: State): Move {
    const move = new Move(this.game.getMoveSize());
    if (state.data[OneDie.kScoreIndex] >= 12 ||
      state.data[OneDie.kTotalScore] + state.data[OneDie.kScoreIndex] > 25) {
      move.data[OneDie.kEndIndex] = 1.0;
    } else {
      move.data[OneDie.kGoIndex] = 1.0;
    }
    return move;
  }
}

File: py/goodStrategy.py
import random

import numpy as np

from oneDie import OneDie


class GoodStrategy:
    game = None

    def __init__(self, g):
        self.game = g

    def getMove(self, state):
        move = np.zeros(self.game.getMoveSize())
        # if (state[OneDie.kScoreIndex] >= 5 or state[OneDie.kTotalScore] > 25):
        # if (state[OneDie.kScoreIndex] >= 12 or state[OneDie.kTotalScore] > 25):
        if random.random() > 0.5:
            move[OneDie.kEndIndex] = 1.0
        else:
            move[OneDie.kGoIndex] = 1.0
        return move
